app.js

```diff
@@ -136,6 +137,32 @@ function addEventListener(indicator, emitter, event, f) {
   });
 }
 
+/*
+  Check couchDbUrl is a valid database URL.
+  @param {string} couchDbUrl - Database URL
+  @param {function(err)} callback - error is undefined if DB exists
+*/
+function exists(db, isBackup, callback) {
+  db.head('', function(err) {
+    err = error.convertResponseError(err, function(err) {
+      if (err && err.statusCode === 404) {
```

Comment:
Do we need to consider 401/403 cases or will these be handled after this check?

Reply:
Good q. The default error factory in the else handles those. The tests that previously asserted 401/403 for _bulk_get have been updated accordingly for the `HEAD` check
i.e. https://github.com/cloudant/couchbackup/blob/212-head-db/test/fatalerrors.js#L128